testsuite/tests/kuadrant/limitador/test_route_rule_subset.py

```diff
@@ -0,0 +1,38 @@
+"""Tests that RLP for a subset of HTTPRouteRule limits all the paths in the HTTPRouteRule"""
+
+import pytest
+
+from testsuite.gateway import RouteMatch, PathMatch, MatchType, HTTPMethod
+from testsuite.policy.rate_limit_policy import Limit, RouteSelector
+
+
+@pytest.fixture(scope="module")
+def route(route, backend):
+    """Add new rule to the route"""
+    route.remove_all_rules()
+    route.add_rule(
+        backend,
+        RouteMatch(path=PathMatch(value="/anything", type=MatchType.PATH_PREFIX), method=HTTPMethod.GET),
+        RouteMatch(path=PathMatch(value="/anything", type=MatchType.PATH_PREFIX), method=HTTPMethod.POST),
+    )
+    return route
+
+
+@pytest.fixture(scope="module")
+def rate_limit(rate_limit):
+    """Add limit to the policy"""
+    selector = RouteSelector(
+        RouteMatch(path=PathMatch(value="/anything", type=MatchType.PATH_PREFIX), method=HTTPMethod.GET)
+    )
+    rate_limit.add_limit("anything", [Limit(5, 10)], route_selectors=[selector])
+    return rate_limit
+
+
+def test_route_rule_subset(client):
+    """Tests that RLP for a subset of HTTPRouteRule applies to the entire HTTPRouteRule"""
```

Comment:
Is that even correct? this feels wrong

Reply:
I had a talk about this with @jsmolar because my first assumption was the exact opposite but looking through the ambiguous wording of [Example 4](https://docs.kuadrant.io/architecture/rfcs/0001-rlp-v2/#example-4-targeting-a-subset-of-an-httprouterule-httproutematch-found) we came to the conclusion that it's supposed to work like that

Reply:
If I remember correctly, we discussed that this is not needed test case and we somehow transformed it into `test_route_different_method.py`, which is correct.